common/alerts.js

```diff
@@ -52,5 +52,21 @@
                 };
             }
         };
-    }]);
+    }])
+    .directive('compile', ['$compile', function ($compile) {
+      return function(scope, element, attrs) {
+        scope.$watch(
+          function(scope) {
+            return scope.$eval(attrs.compile);
```

Comment:
I understand that `attrs.compile` is the html string that you are passing as a variable to the `compile` attribute. 

What is the return value of `$eval` when that value is `null` or empty? Does it make sense to guard against running compile against either of those cases?